layers/cmd/ingest_csv.go

```diff
@@ -195,8 +198,255 @@ func (ci *CSVIngester) Run(pipeline *pipeline.PipelineContext) error {
 	return nil
 }
 
+type CSVJoinIngester struct {
+	BaseIngestParams
+	StartRow     int    `json:"startRow" yaml:"startRow"`
+	EndRow       int    `json:"endRow" yaml:"endRow"`
+	HeaderRow    int    `json:"headerRow" yaml:"headerRow"`
+	ID           string `json:"id" yaml:"id"`
+	IngestByRows bool   `json:"ingestByRows" yaml:"ingestByRows"`
+	Delimiter    string `json:"delimiter" yaml:"delimiter"`
+	ColumnRange  uint   `json:"columnRange" yaml:"columnRange"`
+	initialized  bool
+	ingester     *ls.Ingester
+	entities     []CSVJoinConfig
+}
+
+type CSVJoinConfig struct {
+	VariantID        string
+	StartCol, EndCol int
+	IDCols           []int
+}
+
+type joinData struct {
+	CSVJoinConfig
+	data []string
+	id   string
+}
+
+func (cji *CSVJoinIngester) Run(pipeline *pipeline.PipelineContext) error {
+	var layer *ls.Layer
+	var err error
+	if !cji.initialized {
+		layer, err = LoadSchemaFromFileOrRepo(pipeline.Context, cji.CompiledSchema, cji.Repo, cji.Schema, cji.Type, cji.Bundle)
+		if err != nil {
+			return err
+		}
+		pipeline.Properties["layer"] = layer
+		cji.initialized = true
+		cji.ingester = &ls.Ingester{Schema: layer}
+	}
+
+	parser := csvingest.Parser{
```

Comment:
You need a differnet parser for each entity, with different schemas. Get those variants from bundle.